base_layer/core/src/mempool/pending_pool/pending_pool.rs

```diff
@@ -0,0 +1,239 @@
+        let tx4 = create_test_tx(MicroTari(10_000), MicroTari(200), 2450, 2, 2);
+        let tx5 = create_test_tx(MicroTari(10_000), MicroTari(500), 1000, 3, 3);
+        let tx6 = create_test_tx(MicroTari(10_000), MicroTari(750), 1850, 2, 2);
+
+        let mut pending_pool = PendingPool::new(PendingPoolConfig { storage_capacity: 3 });
+        pending_pool
+            .insert_txs(vec![
+                tx1.clone(),
+                tx2.clone(),
+                tx3.clone(),
+                tx4.clone(),
+                tx5.clone(),
+                tx6.clone(),
+            ])
+            .unwrap();
+        // Check that lowest priority txs were removed to make room for higher priority transactions
+        assert_eq!(pending_pool.len().unwrap(), 3);
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx1.body.kernels[0].excess_sig)
+                .unwrap(),
+            true
+        );
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx2.body.kernels[0].excess_sig)
+                .unwrap(),
+            false
+        );
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx3.body.kernels[0].excess_sig)
+                .unwrap(),
+            true
+        );
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx4.body.kernels[0].excess_sig)
+                .unwrap(),
+            false
+        );
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx5.body.kernels[0].excess_sig)
+                .unwrap(),
+            false
+        );
+        assert_eq!(
+            pending_pool
+                .has_tx_with_excess_sig(&tx6.body.kernels[0].excess_sig)
+                .unwrap(),
+            true
+        );
+
+        assert!(pending_pool.check_status().unwrap());
+    }
+
+    #[test]
+    fn test_remove_unlocked_and_discard_double_spends() {
+        let tx1 = create_test_tx(MicroTari(10_000), MicroTari(500), 500, 2, 1);
+        let tx2 = create_test_tx(MicroTari(10_000), MicroTari(100), 2150, 1, 2);
+        let tx3 = create_test_tx(MicroTari(10_000), MicroTari(1000), 1000, 2, 1);
+        let tx4 = create_test_tx(MicroTari(10_000), MicroTari(200), 2450, 2, 2);
+        let tx5 = create_test_tx(MicroTari(10_000), MicroTari(500), 1000, 3, 3);
+        let tx6 = create_test_tx(MicroTari(10_000), MicroTari(750), 1850, 2, 2);
+
+        let mut pending_pool = PendingPool::new(PendingPoolConfig { storage_capacity: 10 });
+        pending_pool
+            .insert_txs(vec![
+                tx1.clone(),
+                tx2.clone(),
+                tx3.clone(),
+                tx4.clone(),
+                tx5.clone(),
+                tx6.clone(),
+            ])
+            .unwrap();
+        assert_eq!(pending_pool.len().unwrap(), 6);
+
+        let published_block = create_test_block(1500, vec![tx6.clone()]);
```

Comment:
Technically tx6 can't go in this block when its lock height is 1850. The test passes because there's no block validation, which is ok, just pointing that out.